### PR TITLE
docs: DistanceJoint documentation and example

### PR DESCRIPTION
**Pull request by eukleshnin**
# Description

`DistanceJoint` documentation and example

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [-] I have updated/added tests for ALL new/updated/fixed functionality.
- [-] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

PR 2/X for #2005

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: doc/bridge_packages/flame_forge2d/joints.md [hunk 1]
```diff
@@ -60,10 +60,56 @@ It can for example be useful when simulating "soft-bodies".
...
+```
+
+```dart
+    final distanceJointDef = DistanceJointDef()
+      ..initialize(firstBody, secondBody, firstBody.worldCenter, secondBody.worldCenter)
+      ..length = 10
+      ..frequencyHz = 3
+      ..dampingRatio = 0.2;
+
+    world.createJoint(DistanceJoint(distanceJointDef));
```
Comment (spydon) [inline doc/bridge_packages/flame_forge2d/joints.md:94]:
```suggestion
final distanceJointDef = DistanceJointDef()
  ..initialize(firstBody, secondBody, firstBody.worldCenter, secondBody.worldCenter)
  ..length = 10
  ..frequencyHz = 3
  ..dampingRatio = 0.2;

world.createJoint(DistanceJoint(distanceJointDef));
```

File: doc/bridge_packages/flame_forge2d/joints.md [hunk 2]
```diff
@@ -60,10 +60,56 @@ It can for example be useful when simulating "soft-bodies".
...
+```
+
+To create a `DistanceJointDef`, you can use the `initialize` method,
```
Comment (spydon) [inline doc/bridge_packages/flame_forge2d/joints.md:97]: The max line length in the docs is 100, so you don't have to line break until you are closer to that.

File: doc/bridge_packages/flame_forge2d/joints.md [hunk 3]
```diff
@@ -60,10 +60,56 @@ It can for example be useful when simulating "soft-bodies".
...
+which requires two bodies and a world anchor point on each body.
+The definition uses local anchor points, allowing for a slight violation of the constraint
+in the initial configuration. This is useful when saving and loading a game.
+
+The `DistanceJointDef` has three optional parameters that you can set:
+
+- `length`: This parameter determines the distance between
+the two anchor points and must be greater than 0.
```
Comment (spydon) [inline doc/bridge_packages/flame_forge2d/joints.md:105]: Interesting that this one is optional, what happens if it is not defined?

Comment (eukleshnin) [reply doc/bridge_packages/flame_forge2d/joints.md]: It's inited with a default value. Added a note about that.

File: doc/bridge_packages/flame_forge2d/joints.md (hunk 2 above)
Comment (eukleshnin) [reply doc/bridge_packages/flame_forge2d/joints.md:97]: I just feel it's easier to read like that :)

Comment (spydon) [reply doc/bridge_packages/flame_forge2d/joints.md:97]: In some cases it makes sense, but in most it should be closer to 100, to keep consistency with the rest of the docs.

File: doc/bridge_packages/flame_forge2d/joints.md [hunk 4]
```diff
@@ -60,10 +60,56 @@ It can for example be useful when simulating "soft-bodies".
...
+
+To create a `DistanceJointDef`, you can use the `initialize` method,
+which requires two bodies and a world anchor point on each body.
+The definition uses local anchor points, allowing for a slight violation of the constraint
+in the initial configuration. This is useful when saving and loading a game.
+
+The `DistanceJointDef` has three optional parameters that you can set:
+
+- `length`: This parameter determines the distance between
+the two anchor points and must be greater than 0. The default value is 1.
```
Comment (spydon) [inline doc/bridge_packages/flame_forge2d/joints.md:105]: This one for example :)

File: doc/bridge_packages/flame_forge2d/joints.md [hunk 5]
```diff
@@ -35,12 +35,12 @@ Currently, Forge2D supports the following joints:
 ### `ConstantVolumeJoint`
 
 This type of joint connects a group of bodies together and maintains a constant volume within them.
-Essentially, it is a set of `DistanceJoint`s, that connects all bodies one after another.
+Essentially, it is a set of [`DistanceJoint`](#distancejoint)s, that connects all bodies one after another.
```
Comment (spydon) [inline doc/bridge_packages/flame_forge2d/joints.md:38]:
```suggestion
Essentially, it is a set of [`DistanceJoint`](#distancejoint)s, that connects all bodies one after
another.
```
This goes over the max line length

File: packages/flame/README.md [hunk 6]
```diff
@@ -1 +1 @@
-../../README.md
\ No newline at end of file
+../../README.md
```
Comment (spydon) [inline packages/flame/README.md]: This file shouldn't have any changes in it, I'm guessing it is a different type of line break? :thinking:

**Review by spydon — COMMENTED** [review]
Only some small nits, overall it looks good! :)

File: packages/flame/README.md (hunk 6 above)
Comment (eukleshnin) [reply packages/flame/README.md]: Yeah, I reverted it, it was failing the CI. `markdown-check` added it for some reason

Comment (spydon) [reply packages/flame/README.md]: Ah, it might depend on what operating system you run `markdown-fix` on what the results will be

**Review by spydon — APPROVED** [review]

**Pull request merged**